internal/config/transport_test.go

```diff
@@ -56,60 +58,117 @@ func TestRoundTripper_Bind(t *testing.T) {
-		           ForceAttemptHTTP2: false,
-		           },
-		           want: want{},
-		           checkFunc: defaultCheckFunc,
-		       }
-		   }(),
-		*/
-	}
+			return test{
+				name: "return RoundTripper when all parameters are not nil",
+				fields: fields{
+					TLSHandshakeTimeout:   tlsHandshakeTimeout,
+					MaxIdleConns:          maxIdleConns,
+					MaxIdleConnsPerHost:   maxIdleConnsPerHost,
+					MaxConnsPerHost:       maxConnsPerHost,
+					IdleConnTimeout:       idleConnTimeout,
+					ResponseHeaderTimeout: responseHeaderTimeout,
+					ExpectContinueTimeout: expectContinueTimeout,
+					MaxResponseHeaderSize: maxResponseHeaderSize,
+					WriteBufferSize:       writeBufferSize,
+					ReadBufferSize:        readBufferSize,
+					ForceAttemptHTTP2:     forceAttemptHTTP2,
+				},
+				want: want{
+					want: &RoundTripper{
+						TLSHandshakeTimeout:   tlsHandshakeTimeout,
+						MaxIdleConns:          maxIdleConns,
+						MaxIdleConnsPerHost:   maxIdleConnsPerHost,
+						MaxConnsPerHost:       maxConnsPerHost,
+						IdleConnTimeout:       idleConnTimeout,
+						ResponseHeaderTimeout: responseHeaderTimeout,
+						ExpectContinueTimeout: expectContinueTimeout,
+						MaxResponseHeaderSize: maxResponseHeaderSize,
+						WriteBufferSize:       writeBufferSize,
+						ReadBufferSize:        readBufferSize,
+						ForceAttemptHTTP2:     forceAttemptHTTP2,
+					},
+				},
+			}
+		}(),
+		func() test {
+			tlsHandshakeTimeout := "5s"
+			idleConnTimeout := "10s"
+			responseHeaderTimeout := "5s"
+			expectContinueTimeout := "5s"
+			envPrefix := "bind_env_test_rt"
+			m := map[string]string{
+				envPrefix + "tlsHandshakeTimeout":   tlsHandshakeTimeout,
+				envPrefix + "idleConnTimeout":       idleConnTimeout,
+				envPrefix + "responseHeaderTimeout": responseHeaderTimeout,
+				envPrefix + "expectContinueTimeout": expectContinueTimeout,
+			}
 
-	for _, test := range tests {
+			return test{
+				name: "return RoundTripper when the data is loaded environment variable",
+				fields: fields{
+					TLSHandshakeTimeout:   "_" + envPrefix + "tlsHandshakeTimeout_",
+					IdleConnTimeout:       "_" + envPrefix + "idleConnTimeout_",
+					ResponseHeaderTimeout: "_" + envPrefix + "responseHeaderTimeout_",
+					ExpectContinueTimeout: "_" + envPrefix + "expectContinueTimeout_",
+				},
+				want: want{
+					want: &RoundTripper{
+						TLSHandshakeTimeout:   tlsHandshakeTimeout,
+						IdleConnTimeout:       idleConnTimeout,
+						ResponseHeaderTimeout: responseHeaderTimeout,
+						ExpectContinueTimeout: expectContinueTimeout,
+					},
+				},
+				beforeFunc: func(t *testing.T) {
+					t.Helper()
+					for k, v := range m {
+						if err := os.Setenv(k, v); err != nil {
+							t.Fatal(err)
+						}
+					}
+				},
+				afterFunc: func(t *testing.T) {
+					t.Helper()
+					for k := range m {
+						if err := os.Unsetenv(k); err != nil {
+							t.Fatal(err)
+						}
+					}
+				},
+			}
+		}(),
+	}
+	t.Parallel()
```

Comment:
please generate test code using a template.

Reply:
@kevindiu 
or could you plese move this call to the top of the function if it's hard to fix by generation 🤔